src/types/config.d.ts

```diff
@@ -0,0 +1,116 @@
+import AuthSettings from "../core/AuthSettings";
+import { CallingSDKLogger } from "../utils/loggers";
+
+export type VoiceVideoCallingOptionalParams={
+    logger? : CallingSDKLogger | null;
+}
+
+export type LiveChatConfig = {
+    DataMaskingInfo: DataMaskingInfo;
+    LiveChatConfigAuthSettings: AuthSettings;
+    LiveWSAndLiveChatEngJoin: LiveWSAndLiveChatEngJoin;
+    LiveChatVersion: number;
+    ChatWidgetLanguage: ChatWidgetLanguage;
+}
+
+export type ChatWidgetLanguage = {
+    msdyn_localeid: string,
+    msdyn_languagename: string;
+}
+
+export type DataMaskingRules = {
+    [key: string]: string;
+}
+
+export type Setting = {
+    msdyn_maskforcustomer: boolean;
+    msdyn_maskforagent: boolean;
+    "@odata.etag": string;
+}
+
+export type DataMaskingInfo = {
+    dataMaskingRules: DataMaskingRules;
+    setting: Setting;
+}
+
+export type LiveWSAndLiveChatEngJoin = {
```

Comment:
In this type I see many references to | null, is there a reason you've selected this instead of | undefined?

Reply:
I have to confess this was defined by copilot based on the json response from getchatconfig